ast/parser_new.go

```diff
@@ -1301,3 +1405,26 @@ func (p *Parser) setLoc(term *Term, loc *location.Location, offset, end int) *Te
 	}
 	return term
 }
+
+func (p *Parser) validateDefaultRuleValue(rule *Rule) bool {
+	if rule.Head.Value == nil {
+		p.error(rule.Loc(), fmt.Sprintf("default rule must have a value"))
+		return false
+	}
+
+	valid := true
+	vis := NewGenericVisitor(func(x interface{}) bool {
+		switch x.(type) {
+		case *ArrayComprehension, *ObjectComprehension, *SetComprehension: // skip closures
+			return true
+		case Ref, Var:
```

Comment:
Nit: I think we should match calls here so that the error is clearer.

Reply:
Done